node/src/service.rs

```diff
@@ -302,6 +309,56 @@ async fn start_dev_impl(
         telemetry: telemetry.as_mut(),
     })?;
 
+    // Spawning the Blockchain Service if node is running as a Storage Provider.
+    if let Some(provider_options) = provider_options {
+        // File Transfer Service handle is expected to be present when the node is running as a Storage Provider.
+        let file_transfer_service_handle = file_transfer_service_handle.expect(
+            "File Transfer Service handle is expected to be present when the node is running as a Storage Provider. qed",
+        );
```

Comment:
My previous comment refers to this, we never check `.is_some` or `.is_none` with `file_transfer_service_handle`, so why do we make it an `Option`?